runtime/near-vm-logic/src/logic.rs

```diff
@@ -2786,6 +2786,37 @@ impl<'a> VMLogic<'a> {
         }
         Ok(())
     }
+
+    /// Verify an ED25519 signature given a message and a public key.
+    /// # Cost
+    ///
+    /// `base + write_register_base + write_register_byte * num_bytes + ed25519_verify_base + ed25519_verify_byte * num_bytes`
+    pub fn ed25519_verify(
+        &mut self,
+        sig_len: u64,
+        sig_ptr: u64,
+        msg_len: u64,
+        msg_ptr: u64,
+        pub_key_len: u64,
+        pub_key_ptr: u64,
+    ) -> Result<u32> {
+        use ed25519_dalek::{PublicKey, Signature, Verifier};
+
+        self.gas_counter.pay_base(ed25519_verify_base)?;
+        self.gas_counter.pay_per(ed25519_verify_byte, msg_len + sig_len as u64)?;
+        let signature_array = self.get_vec_from_memory_or_register(sig_ptr, sig_len)?;
+        let msg = self.get_vec_from_memory_or_register(msg_ptr, msg_len)?;
+        let pub_key_array = self.get_vec_from_memory_or_register(pub_key_ptr, pub_key_len)?;
+        let pub_key = PublicKey::from_bytes(&pub_key_array)
+            .map_err(|e| VMLogicError::ExternalError(AnyError::new(e.to_string())))?;
+        let signature = Signature::from_bytes(&signature_array)
+            .map_err(|e| VMLogicError::ExternalError(AnyError::new(e.to_string())))?;
```

Comment:
`AnyError` serves completely different purpose, and that's the reason why no other function here uses that. Follow the pattern elsewhere: add a new variant to `HostError`.